packages/headless/src/controllers/triggers/headless-notify-trigger.ts

```diff
@@ -0,0 +1,76 @@
+import {SearchEngine} from '../../app/search-engine/search-engine';
+import {ConfigurationSection, TriggerSection} from '../../state/state-sections';
+import {configuration, triggers} from '../../app/reducers';
+import {buildController, Controller} from '../controller/headless-controller';
+import {loadReducerError} from '../../utils/errors';
+import {logNotifyTrigger} from '../../features/triggers/trigger-analytics-actions';
+
+/**
+ * The `NotifyTrigger` controller handles Notify triggers.
+ */
+export interface NotifyTrigger extends Controller {
+  /**
+   * the state of the `NotifyTrigger` controller.
+   */
+  state: NotifyTriggerState;
+}
+
+/**
+ * A scoped and simplified part of the headless state that is relevant to the `NotifyTrigger` controller.
+ */
+export interface NotifyTriggerState {
+  /**
+   * The notification to present to the user after receiving a notification trigger.
+   */
+  notify: string;
+}
+
+/**
+ * Creates a `NotifyTrigger` controller instance.
+ *
+ * @param engine - The headless engine.
+ * @returns A `NotifyTrigger` controller instance.
+ * */
+export function buildNotifyTrigger(engine: SearchEngine): NotifyTrigger {
+  if (!loadNotifyTriggerReducers(engine)) {
+    throw loadReducerError;
+  }
+
+  const controller = buildController(engine);
+  const {dispatch} = engine;
+
+  const getState = () => engine.state;
+
+  let previousNotify: string = getState().triggers.notify;
+
+  return {
+    ...controller,
+
+    subscribe(listener: () => void) {
+      const strictListener = () => {
+        const hasChanged = previousNotify !== this.state.notify;
+        previousNotify = this.state.notify;
```

Comment:
You may need to declare `previousNotify` inside this `subscribe` block. Otherwise, subscribing multiple times to a notify trigger controller would probably not call all subscribers.

Reply:
If I declare it in the subscribe block, I'm not sure how we would be able to detect when `this.state.notification`'s value changed. It would then be reinitialized whenever subscribe is called so it would always have the new value from the recently updated state and there would be no way of keeping track of the previous value.